index.js

```diff
@@ -1,87 +1,72 @@
-'use strict'
+const bcrypt = require('bcrypt');
 
-const Bcrypt = require('bcrypt')
+const RECOMMENDED_ROUNDS = 12;
+const BCRYPT_HASH_REGEX = /^\$2[ayb]\$[0-9]{2}\$[A-Za-z0-9./]{53}$/;
 
-const RECOMMENDED_ROUNDS = 12
-
-const REGEXP = /^\$2[ayb]\$[0-9]{2}\$[A-Za-z0-9./]{53}$/
-
-module.exports = (options) => {
+// Plugin that provides automatic bcrypt hashing for passwords on Objection.js models.
+const objectionPassword = (options) => {
   // Provide good defaults for the options if possible.
-  options = Object.assign({
+  options = {
     allowEmptyPassword: false,
     passwordField: 'password',
-    rounds: RECOMMENDED_ROUNDS
-  }, options)
+    rounds: RECOMMENDED_ROUNDS,
+    ...options
+  };
 
-  // Return the mixin. If your plugin doesn't take options, you can simply export
-  // the mixin. The factory function is not needed.
+  // Return the mixin.
+  // If the plugin doesn't take options, the mixin can be exported directly. The factory function is not needed.
   return (Model) => {
     return class extends Model {
-      $beforeInsert (context) {
-        const maybePromise = super.$beforeInsert(context)
+      async $beforeInsert(...args) {
```

Comment:
I also prefer async/await too.

 think I've seen `bluebird` used to handle Promise vs async in libraries before, but don't have experience with it myself.

Not needed if you do the major version bump as suggested in [this thread](https://github.com/scoutforpets/objection-password/pull/32/files#r481586422) though I suppose.

Reply:
I'm thinking the right thing to do here is continue forward with all of the suggested changes, bump the version to 3.x, but also update the 2.x branch with `bcrypt@5.x`. Otherwise, for those who aren't on a newer version of Node won't be able to get the newer, more secure version of bcrypt without updating node.

Reply:
Makes sense to me, good call thinking of those on older versions.

Sounds like the PR needs to be split up then

Reply:
As we discussed in the other comment chain, bcrypt 5.x requires node 12+ so either way people on older versions of this library would have to upgrade their version of node to address the bcrypt vulnerability.

Given this, I don't see a reason to create another revision for the 2.x _branch_

Reply:
Yep, 100% correct...evidently I needed more coffee before I wrote that 😆